lib/init/features/project.js

```diff
@@ -11,32 +11,49 @@ var utils = require("../../utils");
 var NO_PROJECT = "[don't setup a default project]";
 var NEW_PROJECT = "[create a new project]";
 
-module.exports = function(setup, config) {
-  setup.project = {};
-
-  logger.info();
-  logger.info("First, let's associate this project directory with a Firebase project.");
-  logger.info(
-    "You can create multiple project aliases by running " + clc.bold("firebase use --add") + ", "
-  );
-  logger.info("but for now we'll just set up a default project.");
-  logger.info();
+/**
+ * Get the user's desired project, prompting if necessary.
+ * Returns an object with three fields:
+ *
+ * {
+ *  id: project ID [required]
+ *  name: project display name [optional]
+ *  instance: project database instance [optional]
+ * }
+ */
+function _getProject(options) {
+  // The user passed in a --project flag directly, so no need to
+  // load all projects.
+  if (options.project) {
+    return api
+      .request("GET", "/v1beta1/projects/" + options.project, {
+        auth: true,
+        origin: api.firebaseApiOrigin,
+      })
+      .then(function(response) {
+        // TODO: CATCH
+        return {
+          id: response.body.projectId,
+          name: response.body.displayName,
+          instance: response.body.resources.realtimeDatabaseInstance,
+        };
+      })
+      .catch(function(e) {
+        return utils.reject("Error getting project " + options.project, { original: e });
+      });
+  }
 
+  // Load all projects and prompt the user to choose.
   return api.getProjects().then(function(projects) {
     var choices = _.map(projects, function(info, projectId) {
       return {
         name: projectId,
         label: info.name + " (" + projectId + ")",
       };
     });
+    choices = _.orderBy(choices, ["name"], ["asc"]);
```

Comment:
Should we order by name, or by project id? Project id is the one that matters. I may be biased because I wish `name` just didn't exist.

Reply:
Resolved.  Note that `name` here is the project ID.